common/transport/mqtt/src/main/java/org/thingsboard/server/transport/mqtt/gateway/GatewayMetricsService.java

```diff
@@ -0,0 +1,120 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.thingsboard.server.transport.mqtt.gateway;
+
+import jakarta.annotation.PostConstruct;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+import org.thingsboard.common.util.JacksonUtil;
+import org.thingsboard.server.common.data.id.DeviceId;
+import org.thingsboard.server.common.transport.TransportService;
+import org.thingsboard.server.common.transport.TransportServiceCallback;
+import org.thingsboard.server.gen.transport.TransportProtos;
+import org.thingsboard.server.queue.scheduler.SchedulerComponent;
+import org.thingsboard.server.transport.mqtt.TbMqttTransportComponent;
+import org.thingsboard.server.transport.mqtt.gateway.metrics.GatewayMetricsData;
+import org.thingsboard.server.transport.mqtt.gateway.metrics.GatewayMetricsState;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.TimeUnit;
+
+@Slf4j
+@Service
+@TbMqttTransportComponent
+public class GatewayMetricsService {
+
+    public static final String GATEWAY_METRICS = "gatewayMetrics";
+
+    @Value("${transport.mqtt.gateway_metrics_report_interval_sec:3600}")
+    private int metricsReportIntervalSec;
+
+    @Autowired
+    private SchedulerComponent scheduler;
+
+    @Autowired
+    private TransportService transportService;
+
+    private Map<DeviceId, GatewayMetricsState> states = new ConcurrentHashMap<>();
+
+    @PostConstruct
+    private void init() {
+        scheduler.scheduleAtFixedRate(this::reportMetrics, metricsReportIntervalSec, metricsReportIntervalSec, TimeUnit.SECONDS);
+    }
+
+    public void process(TransportProtos.SessionInfoProto sessionInfo, DeviceId gatewayId, Map<String, GatewayMetricsData> data, long ts) {
+        states.computeIfAbsent(gatewayId, k -> new GatewayMetricsState(sessionInfo)).update(ts, data);
+    }
+
+    public void onDeviceUpdate(TransportProtos.SessionInfoProto sessionInfo, DeviceId gatewayId) {
+        var state = states.get(gatewayId);
+        if (state != null) {
+            state.updateSessionInfo(sessionInfo);
+        }
+    }
+
+    public void onDeviceDelete(DeviceId deviceId) {
+        states.remove(deviceId);
+    }
+
+    public void onDeviceDisconnect(DeviceId deviceId) {
+        GatewayMetricsState state = states.remove(deviceId);
+        if (state != null) {
+            reportMetrics(state, System.currentTimeMillis());
+        }
+    }
+
+    public void reportMetrics() {
+        if (states.isEmpty()) {
```

Comment:
same error as we faced with device state service. In case of any exceptions, this process will stop working forever.